sdk/src/main/scala/com/horizen/forge/ForgingControl.scala

```diff
@@ -0,0 +1,90 @@
+package com.horizen.forge
+
+import java.time.Instant
+import java.util.concurrent.Executors
+
+import akka.actor.{Actor, ActorRef, ActorSystem, Props}
+import akka.util.Timeout
+import com.horizen._
+import com.horizen.block.SidechainBlock
+import com.horizen.consensus.TimeToEpochSlotConverter
+import com.horizen.forge.Forger.SendMessages.{ForgeResult, ForgeSuccess}
+import com.horizen.params.NetworkParams
+import scorex.core.NodeViewHolder.ReceivableMessages.LocallyGeneratedModifier
+import scorex.util.ScorexLogging
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration.FiniteDuration
+import scala.util.Success
+
+class ForgingControl(settings: SidechainSettings, forgerRef: ActorRef, viewHolderRef: ActorRef, val params: NetworkParams) extends Actor with ScorexLogging with TimeToEpochSlotConverter {
+  val timeoutDuration: FiniteDuration = settings.scorexSettings.restApi.timeout
+  implicit val timeout: Timeout = Timeout(timeoutDuration)
+
+  @volatile private var forgingIsActive: Boolean = true
+
+  val consensusMillisecondsInSlot: Int = params.consensusSecondsInSlot * 1000
+  val forgingInitiator: Runnable = () => {
+    while (true) {
+      if (forgingIsActive) {
+        val currentTime: Long = Instant.now.getEpochSecond
+        log.debug(s"Send TryForgeNextBlockForEpochAndSlot message where epoch number is ${timeStampToEpochNumber(currentTime)} and slot number is ${timeStampToSlotNumber(currentTime)}")
+        forgerRef ! Forger.ReceivableMessages.TryForgeNextBlockForEpochAndSlot(timeStampToEpochNumber(currentTime), timeStampToSlotNumber(currentTime))
+      }
+      Thread.sleep(consensusMillisecondsInSlot)
```

Comment:
It can be a situation on Node first run, automatic forging starts and fails, because `currentTime` belongs to the same slot as genesisBlock.
Also, this loop is executed even if forging was stopped.